akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala

```diff
@@ -317,6 +318,96 @@ class CircuitBreaker(
   }
 
   /**
+   * Adds a callback to execute when call finished with success.
+   *
+   * The callback is run in the [[scala.concurrent.ExecutionContext]] supplied in the constructor.
+   *
+   * @param callback Handler to be invoked on successful call, where passed value is elapsed time in nanoseconds.
+   * @return CircuitBreaker for fluent usage
+   */
+  def onCallSuccess(callback: Long ⇒ Unit): CircuitBreaker = onCallSuccess(new Consumer[Long] {
+    def accept(result: Long): Unit = callback(result)
+  })
+
+  /**
+   * JavaAPI for onCallSuccess
+   *
+   * @param callback Handler to be invoked on successful call, where passed value is elapsed time in nanoseconds.
+   * @return CircuitBreaker for fluent usage
+   */
+  def onCallSuccess(callback: Consumer[Long]): CircuitBreaker = {
```

Comment:
Just double checked with Scala 2.12 and Java 8. These are not usable with Java 8 lambdas. Javac throws this error:
```
[error] master/akka-actor-tests/src/test/java/akka/pattern/CircuitBreakerTest.java:36: reference to onCallSuccess is ambiguous
[error]   both method onCallSuccess(scala.Function1<java.lang.Object,scala.runtime.BoxedUnit>) in akka.pattern.CircuitBreaker and method onCallSuccess(java.util.function.Consumer<java.lang.Object>) in akka.pattern.CircuitBreaker match
[error]     breaker.onCallSuccess(x -> System.out.println(x));
[error] master/akka-actor-tests/src/test/java/akka/pattern/CircuitBreakerTest.java:36:  bad return type in lambda expression
[error]     void cannot be converted to scala.runtime.BoxedUnit
[error]     breaker.onCallSuccess(x -> System.out.println(x));
```

Don't know any other way than to have different names for the Java and Scala methods.